### PR TITLE
Add option to the keycloak_client module

### PR DESCRIPTION
**Pull request by Gaetan2907**
##### SUMMARY
Add `authentication_flow_binding_overrides` option to the keycloak_client module.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
`keycloak_client.py`

##### ADDITIONAL INFORMATION
It is now possible to use the `authentication_flow_binding_overrides` option, like on the following example: 

```
- name: Create or update a Keycloak client (with authentication_flow_binding_override)
  community.general.keycloak_client:
    auth_keycloak_url: https://auth.example.com/auth
    token: "{{ access_token }}"
    state: present
    realm: master
    client_id: test
    authentication_flow_binding_override:
        browser: 4c90336b-bf1d-4b87-916d-3677ba4e5fbb
```
New feature tested on Keycloak 8.0.1.

### CODE REVIEW

File: changelogs/fragments/2949-add_authentication-flow-binding_keycloak-client.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - keycloak_client - add authentication_flow_binding_overrides option to the keycloak_client module (https://github.com/ansible-collections/community.general/pull/2949).
```
Comment (aminvakil) [inline changelogs/fragments/2949-add_authentication-flow-binding_keycloak-client.yml:3]:
```suggestion
  - keycloak_client - add ``authentication_flow_binding_overrides`` option (https://github.com/ansible-collections/community.general/pull/2949).
```

File: plugins/modules/identity/keycloak/keycloak_client.py [hunk 2]
```diff
@@ -318,6 +318,13 @@
         aliases:
             - authorizationSettings
 
+    authentication_flow_binding_overrides:
+        description:
+            - Override realm authentication flow bindings.
+        type: dict
+        aliases:
+            - authenticationFlowBindingOverrides
```
Comment (aminvakil) [inline plugins/modules/identity/keycloak/keycloak_client.py:326]:
```suggestion
            - authenticationFlowBindingOverrides
        version_added: 3.4.0
```

**Review by aminvakil — CHANGES_REQUESTED** [review]
Thanks for adding this!
Maybe you forgot to push the actual code that does this change? :)

**Comment by Gaetan2907** [comment]
> Maybe you forgot to push the actual code that does this change? :)

No more code need to be added, the parameter is automatically included in the API call

**Comment by aminvakil** [comment]
> > Maybe you forgot to push the actual code that does this change? :)
> 
> No more code need to be added, the parameter is automatically included in the API call

Right, I didn't know about the magic this module does! ;)

File: plugins/modules/identity/keycloak/keycloak_client.py [hunk 3]
```diff
@@ -593,6 +601,8 @@
     default_roles:
       - test01
       - test02
+    authentication_flow_binding_override:
```
Comment (aminvakil) [inline plugins/modules/identity/keycloak/keycloak_client.py:604]:
```suggestion
    authentication_flow_binding_overrides:
```

Comment (Gaetan2907) [reply plugins/modules/identity/keycloak/keycloak_client.py:604]: Nice catch ! :)

**Review by aminvakil — CHANGES_REQUESTED** [review]
LGTM
I'm just blocking this from getting merged until we decide whether we want to release 3.3.2 or not.

**Review by felixfontein — COMMENTED** [review]
It would be nice to have a test for this new parameter.

**Review by felixfontein — APPROVED** [review]

**Review by aminvakil — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@Gaetan2907 thanks for implementing this!
@aminvakil thanks for reviewing!

**Pull request merged**